Marlin/src/HAL/HAL_LPC1768/LPC1768_PWM.cpp

```diff
@@ -96,8 +96,8 @@ typedef struct {            // holds all data needed to control/init one of the
   uint8_t             servo_index;    // 0 - MAX_SERVO -1 : servo index,  0xFF : PWM channel
 } PWM_map;
 
-PWM_map PWM1_map_A[NUM_ISR_PWMS] = {0};
-PWM_map PWM1_map_B[NUM_ISR_PWMS] = {0};;
+PWM_map PWM1_map_A[NUM_ISR_PWMS] = { 0 };
```

Comment:
I'm not sure what my own feelings are on this, so this is a question more than a correction. Using `type array[size] = { 0 };` although looking the best leads to people thinking that the initialiser lists works in a way that they don't, because ` = { 1 };` would not initialise an array to all ones, only the first element would be one the rest defaulted to zero, .. so would it be better to use the ` = {};` notation as it means default the structure elements to zero, or the ` = { 0 };` notation because the intention is clearer even if the notation is not.

Reply:
Agreed.   But what is the syntax if I wanted every element set to 1?   Can I do a 
```
type array[size] = 1;
```
Some how I need a way to set every element of the array to a known value.

Reply:
There is no syntax in standard C++ that lets an initialiser list do that I'm afraid, your stuck with the normal memset, loops ect. 

This is not available in c++, so useless to Marlin, but as an side note C99 supports the [syntax](https://gcc.gnu.org/onlinedocs/gcc/Designated-Inits.html) `= {[0] = 1, [5] = 2, [7] = 3}` which gcc has an extension for allowing `type array[size] = {[0 ... size] = 1}`, the syntax has the hidden memory cost of size * sizeof(type) though so even though it exists shouldn't really be used.

Reply:
I'll use the {0,0,0} format to set the table elements when needed and depend on the compiler to initialize the arrays to all zeros.